### PR TITLE
Use bundled packages and update CI for publishing

### PR DESCRIPTION
**Pull request by chlunde**
### Description of your changes

Based on work by @jastang in another repo.

Updates the build machinery to build bundled provider packages, as well as publish to crossplane-contrib OCI repos in both DockerHub and Upbound registries.

This is based on the branch for #115 which should be merged first.

xref crossplane/crossplane#3350

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

```
./.cache/tools/linux_x86_64/up-v0.13.0 repository create provider-sql
chlunde/provider-sql created
VERSION=v0.6.0 make build.all
XPKG_REG_ORGS=xpkg.upbound.io/chlunde BRANCH_NAME=master VERSION=v0.6.0 make publish
```
[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by chlunde** [comment]
@jastang do you see what is missing here? Btw. do we need any secrets on the repo?

**Review by hasheddan — COMMENTED** [review]
@chlunde thanks for putting together! We'll need to do some image consolidation as well -- https://github.com/crossplane-contrib/provider-sql/tree/master/cluster/images/provider-sql-controller can become just `provider-sql` 👍🏻

**Comment by chlunde** [comment]
Thanks @hasheddan - I didn't notice that!
Publish tested OK:
```
./.cache/tools/linux_x86_64/up-v0.13.0 repository create provider-sql
chlunde/provider-sql created
VERSION=v0.6.0 make build.all
XPKG_REG_ORGS=xpkg.upbound.io/chlunde BRANCH_NAME=master VERSION=v0.6.0 make publish
```

To fix e2e, I had to remove this part from integration_test.sh, even though it is still in provider-kafka:

https://github.com/crossplane-contrib/provider-kafka/blob/main/cluster/local/integration_tests.sh#L90-L93

**Review by Duologic — APPROVED** [review]
@hasheddan Can you have another look and perhaps merge this?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -299,14 +300,22 @@ jobs:
...
+          username: ${{ secrets.CONTRIB_DOCKER_USR }}
+          password: ${{ secrets.CONTRIB_DOCKER_PSW }}
+
+      - name: Login to Upbound
+        uses: docker/login-action@v1
+        if: env.XPKG_ACCESS_ID != ''
+        with:
+          registry: xpkg.upbound.io
+          username: ${{ secrets.XPKG_ACCESS_ID }}
+          password: ${{ secrets.XPKG_TOKEN }}
```
Comment (jastang) [inline .github/workflows/ci.yml:314]: @chlunde @Duologic I don't have admin access to this repo but have these been configured already? I don't see them on the Marketplace (in the `crossplane-contrib` org), so I think we'll have to create the robot token there and then update these.

**Review by jastang — COMMENTED** [review]
Left a comment re: configuring the robot token for `xpkg.upbound.io` to publish under `crossplane-contrib`

**Comment by Duologic** [comment]
@chlunde and @hasheddan Can you have another look at this PR?

File: Makefile [hunk 2]
```diff
@@ -6,26 +6,71 @@ PROJECT_REPO := github.com/crossplane-contrib/$(PROJECT_NAME)
...
 -include build/makelib/golang.mk
 
+# kind-related versions
+KIND_VERSION ?= v0.12.0
+KIND_NODE_IMAGE_TAG ?= v1.23.4
+
+# ====================================================================================
 # Setup Kubernetes tools
+
+UP_VERSION = v0.13.0
```
Comment (jastang) [inline Makefile:39]:
```suggestion
UP_VERSION = v0.18.0
```

can we bump the `build` submodule in tandem (to a6e25af)? it's behind

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jbw976) [reply .github/workflows/ci.yml:314]: @jastang has created the robot ID/token and I have added it as a new repository secret that should be available to this github action ✅

**Comment by Duologic** [comment]
ugh, can't sign that -_-

**Comment by Duologic** [comment]
Merged through #153

**Pull request closed without merging**